pkg/certificate/authorization/endpoint.go

```diff
@@ -0,0 +1,113 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package authorization
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"path"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+// ErrEndpoint is the default error class for the authorization endpoint.
+var ErrEndpoint = errs.Class("authorization endpoint error")
+
+// Endpoint provides a http endpoint for interacting with an authorization service.
+type Endpoint struct {
+	log      *zap.Logger
+	db       *DB
+	service  *Service
+	server   http.Server
+	listener net.Listener
+}
+
+// NewEndpoint creates a authorization endpoint.
+func NewEndpoint(log *zap.Logger, db *DB, listener net.Listener) *Endpoint {
+	service := NewService(log, db)
```

Comment:
Elsewhere in the codebase our endpoints are largely stateless and don't create or own objects like this *Service. Can we make it so the Service is created in peer.go (or wherever needs to call NewEndpoint()) and the Service pointer gets passed in to NewEndpoint() as a parameter?